apps/mobile/src/components/TrackCard/TrackCard.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback } from 'react'
+import { View, Text, Pressable, Alert } from 'react-native'
+import { useTranslation } from 'react-i18next'
+import { Eye, EyeOff, Settings, Users, UserCheck } from 'lucide-react-native'
+import clsx from 'clsx'
+import Button from 'components/Button'
+import useCurrentGroup from '@hylo/hooks/useCurrentGroup'
+import useHasResponsibility from '@hylo/hooks/useHasResponsibility'
+import useOpenURL from 'hooks/useOpenURL'
+import { groupUrl } from 'util/navigation'
+import { RESP_MANAGE_TRACKS } from 'store/constants'
+
+function TrackCard({ track, groupSlug }) {
+  const { t } = useTranslation()
+  const [{ currentGroup }] = useCurrentGroup()
+  const hasResponsibility = useHasResponsibility({ forCurrentGroup: true, forCurrentUser: true })
+  const canEdit = hasResponsibility(RESP_MANAGE_TRACKS)
+  const openURL = useOpenURL()
+
+  const handlePublishTrack = useCallback((publishedAt) => {
+    Alert.alert(
+      t('Confirm'),
+      publishedAt 
+        ? t('Are you sure you want to publish this track?') 
+        : t('Are you sure you want to unpublish this track?'),
+      [
+        {
+          text: t('Cancel'),
+          style: 'cancel'
+        },
+        {
+          text: t('OK'),
+          onPress: () => {}
+        }
+      ]
+    )
+  }, [track.id])
+
+  const { actionDescriptor, actionDescriptorPlural, name, numActions, numPeopleCompleted, numPeopleEnrolled, publishedAt, didComplete, isEnrolled } = track
+
+  const navigateToTrack = () => {
+    const trackUrl = `${groupUrl(groupSlug || currentGroup?.slug, 'tracks')}/${track.id}`
+
+    openURL(trackUrl)
+  }
+
+  return (
+    <Pressable 
+      onPress={navigateToTrack}
+      className='mb-2'
+    >
+      <View className='rounded-xl p-2 flex flex-col bg-card/50 border-2 border-card/30'>
+        <View className='flex flex-row justify-between items-center pb-1'>
+          <Text className='text-base text-foreground font-medium' numberOfLines={1}>{name}</Text>
+          <Text className='text-xs text-foreground/60 ml-2'>
+            {t('{{num}} {{actionName}}', { num: numActions, actionName: numActions === 1 ? actionDescriptor : actionDescriptorPlural })}
+          </Text>
+        </View>
+
+        <View className='flex-row justify-between items-center'>
+          {canEdit && (
+            <View className='flex-row items-center gap-x-2 bg-input p-2 rounded-md'>
+              {/* <Button
```

Comment:
If we want to allow admins to publish or unpublish on mobile....

Reply:
I could see that for sure, but lets start without any admin stuff on mobile, just to make it simple